gtk4/src/tree_model_filter.rs

```diff
@@ -18,3 +19,57 @@ impl TreeModelFilter {
         }
     }
 }
+
+pub trait TreeModelFilterExtManual: 'static {
+    #[doc(alias = "gtk_tree_model_filter_set_modify_func")]
+    fn set_modify_func<F: Fn(&TreeModel, &TreeIter, i32) -> glib::Value + 'static>(
+        &self,
+        types: &[glib::Type],
+        func: F,
+    );
+}
+
+impl<O: IsA<TreeModelFilter>> TreeModelFilterExtManual for O {
+    #[doc(alias = "gtk_tree_model_filter_set_modify_func")]
+    fn set_modify_func<F: Fn(&TreeModel, &TreeIter, i32) -> glib::Value + 'static>(
+        &self,
+        types: &[glib::Type],
+        func: F,
+    ) {
+        unsafe {
+            let types_ptr: Vec<glib::ffi::GType> = types.iter().map(|t| t.to_glib()).collect();
+
+            unsafe extern "C" fn func_trampoline<
+                F: Fn(&TreeModel, &TreeIter, i32) -> glib::Value + 'static,
+            >(
+                model: *mut ffi::GtkTreeModel,
+                iter: *mut ffi::GtkTreeIter,
+                value: *mut glib::gobject_ffi::GValue,
+                column: i32,
+                user_data: glib::ffi::gpointer,
+            ) {
+                let f: &F = &*(user_data as *const F);
+                let ret = f(&from_glib_borrow(model), &from_glib_borrow(iter), column);
+                *value = *ret.to_glib_none().0;
```

Comment:
This looks wrong memory management-wise. Shouldn't the return value be `transfer full` and owned by the caller? If so, you want `*value = ret.into_raw();`

I guess we also want an `into_raw()` for `GError`...

Reply:
> This looks wrong memory management-wise. Shouldn't the return value be transfer full and owned by the caller? If so, you want *value = ret.into_raw();

The value is annotated as out caller allocates, I have changed it to use `into_raw`.